windowing/gbm: hide and disable setting Use Display As Clock

HiassofT opened the pull request:
## Description
Use display as clock is not supported on GBM and before the
multi-windowing change it has been hidden in settings via gbm.xml.

Hide it in settings so it can't be enabled in player settings and
also make sure it's disabled in case it had been enabled before,
otherwise users would be left without audio passthrough and no
option to fix this via the GUI.

## Motivation and Context
Restore behaviour prior to the multi-windowing change and make sure
users aren't able to select a known non-working option

## How Has This Been Tested?
Runtime tested on LibreELEC RPi4, checked that previously enabled setting
is getting disabled (with a WARN in log) and that option is not shown in video
player settings.

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

lrusak commented on xbmc/windowing/gbm/WinSystemGbm.cpp:
```diff
@@ -85,6 +85,18 @@ bool CWinSystemGbm::InitWindowSystem()
+  {
+    CServiceBroker::GetSettingsComponent()
+        ->GetSettings()
+        ->SetBool(CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK, false);
```
It should also be `SetVisible(false)`

lrusak commented on xbmc/windowing/gbm/WinSystemGbm.cpp:
```diff
@@ -85,6 +85,18 @@ bool CWinSystemGbm::InitWindowSystem()
+    CServiceBroker::GetSettingsComponent()
+        ->GetSettings()
+        ->SetBool(CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK, false);
+    CLog::Log(LOGWARNING, "CWinSystemGbm::%s: Setting '%s' is not supported and has been disabled",
```
no need for the log really

HiassofT replied:
OK, will drop it and replace by an unconditional set(false)

lrusak commented on xbmc/windowing/gbm/WinSystemGbm.cpp:
```diff
@@ -85,6 +85,18 @@ bool CWinSystemGbm::InitWindowSystem()
       ->GetSettings()
       ->GetSetting("videoscreen.limitguisize")
       ->SetVisible(true);
+  CServiceBroker::GetSettingsComponent()
```
can you please refactor this with the above code to something like:
```
auto settingsComponent = CServiceBroker::GetSettingsComponent();
if (!settingsComponent)
  return false;

auto settings = settingsComponent->->GetSettings();
if (!settings)
  return false;

auto setting = settings->GetSetting(CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK);
if (setting)
  setting->SetVisible(false);
```

HiassofT replied:
makes sense, will do!

HiassofT committed:
windowing/gbm: add null checks before accessing settings

Signed-off-by: Matthias Reichl <hias@horus.com>

HiassofT committed:
windowing/gbm: hide and disable setting Use Display As Clock

Use display as clock is not supported on GBM and before the
multi-windowing change it has been hidden in settings via gbm.xml.

Hide it so it can't be enabled in videoplayer settings and
also disable it - otherwise users who previously enabled it
would be left without audio passthrough and no option to fix
this via the GUI.

Signed-off-by: Matthias Reichl <hias@horus.com>

HiassofT commented:
updated with suggested changes

lrusak commented:
jenkins build this please

lrusak commented:
@DaveTBlake it's your button :smile:

HeresJonny commented:
AFAIK, the speedup functionality https://forum.kodi.tv/showthread.php?tid=10023&pid=2379972#pid2379972 depends on "Sync playback to display". This is one of THE killer functions within Kodi for me to watch movies and TV episodes faster. Please be sure to provide an alternative to keep it working.

lrusak commented:
> AFAIK, the speedup functionality https://forum.kodi.tv/showthread.php?tid=10023&pid=2379972#pid2379972 depends on "Sync playback to display". This is one of THE killer functions within Kodi for me to watch movies and TV episodes faster. Please be sure to provide an alternative to keep it working.

Please don't spam old PR's. If you have a concern please bring it up in the forum thanks.

HeresJonny commented:
Well, 2 days is certainly far from old, and you have taken note of it so my work is done. I don't want to waste your time by having you read this twice.
